compiler/rustc_macros/src/current_version.rs

```diff
@@ -42,8 +19,8 @@ struct RustcVersion {
 }
 
 impl RustcVersion {
-    fn parse_env_var(env_var: &LitStr) -> Result<Self, Box<dyn std::error::Error>> {
-        let value = proc_macro::tracked_env::var(env_var.value())?;
+    fn parse_cfg_release() -> Result<Self, Box<dyn std::error::Error>> {
+        let value = proc_macro::tracked_env::var("CFG_RELEASE")?;
```

Comment:
this error will be quite bad and just say not found. With the old syntax it was at least clear from the source code. Can you print a useful error message instead in the code above?

Reply:
I tweaked it, it now says this if the variable isn't found (I temporarily changed the name to `CFG_RELEASE2` to trigger the error):
```
error: CFG_RELEASE2 env var: environment variable not found
  --> compiler/rustc_session/src/version.rs:12:31
   |
12 |     pub const CURRENT: Self = current_rustc_version!();
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^ in this macro invocation
   |
  ::: /home/njn/dev/rust3/compiler/rustc_macros/src/lib.rs:33:1
   |
33 | pub fn current_rustc_version(input: TokenStream) -> TokenStream {
   | --------------------------------------------------------------- in this expansion of `current_rustc_version!`
```
Compared to the old code:
```
error: environment variable not found
  --> compiler/rustc_session/src/version.rs:12:31
   |
12 |     pub const CURRENT: Self = current_rustc_version!(env!("CFG_RELEASE2"));
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ in this macro invocation
   |
  ::: /home/njn/dev/rust2/compiler/rustc_macros/src/lib.rs:30:1
   |
30 | pub fn current_rustc_version(input: TokenStream) -> TokenStream {
   | --------------------------------------------------------------- in this expansion of `current_rustc_version!`
```
So the variable name is shown in the first line of the error message instead of in the context. Ok?